### PR TITLE
refactor: Migrate Settings from XML to compose 

### LINKED ISSUE #2570: refactor: Migrate Settings from XML to compose

### PR DESCRIPTION
**Pull request by itsPronay**
Fixes #2570 

xml | compose 
------|--------------
image | image

[Screencast_06_04_2024_03:29:22_AM.webm](https://github.com/openMF/mifos-mobile/assets/78101731/806088ae-a34d-4d8c-9f76-fca7fa90f3dc)

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by therajanmaurya** [comment]
image
Give 16dp end padding to update endpoint description

File: app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+                val selectedLanguageValue =
+                    resources.getStringArray(R.array.languages_value)[selectedLanguageIndex]
+
+                val sharedPreferences =
+                    PreferenceManager.getDefaultSharedPreferences(requireContext())
+                sharedPreferences.edit()
+                    .putString(getString(R.string.language_type), selectedLanguageValue).apply()
+
+                dialog.dismiss()
+            }.show()
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt]: Migrate Alert dialogs as well in compose

File: app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                selectedLanguageValue = "System_Language"
+            } else selectedLanguageValue = "en"
+        }
+        return languageValuesArray.indexOf(selectedLanguageValue)
+    }
+
+    private fun changeTheme() {
+        val previouslySelectedTheme = prefsHelper.appTheme
+
+        MaterialAlertDialogBuilder(requireContext()).setTitle(getString(R.string.change_app_theme))
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt]: migrate dialog to compose

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt (hunk 1 above)
Comment (itsPronay) [reply app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt]: done

File: app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            }
+        }
+    }
+
+
+    private fun changeLanguage() {
+        withMutableSnapshot {
+            viewModel.invokeLanguageUpdate = true
+        }
+    }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt]: do this in compose screen.

File: app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+    private fun updateLanguage(language: String) {
+        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(requireContext())
+        sharedPreferences.edit().putString(getString(R.string.language_type), language).apply()
+    }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt]: update shared prefs from compose screen only

File: app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+            prefsHelper.getString(getString(R.string.language_type), null)
+        val languageValuesArray = resources.getStringArray(R.array.languages_value)
+
+        if (!(languageValuesArray.contains(selectedLanguageValue))) {
+            if (languageValuesArray.contains(Locale.getDefault().language)) {
+                selectedLanguageValue = "System_Language"
+            } else selectedLanguageValue = "en"
+        }
+        return languageValuesArray.indexOf(selectedLanguageValue)
+    }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt]: in compose screen

File: app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+        withMutableSnapshot {
+            viewModel.invokeThemeUpdate = true
+        }
+    }
+
+    private fun updateEndpoint() {
+        withMutableSnapshot {
+            viewModel.invokeEndpointUpdate = true
+        }
+    }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/settings/SettingsFragment.kt]: in compose screen

File: app/src/main/java/org/mifos/mobile/ui/settings/SettingsScreen.kt [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+    handleEndpointupdate: (etBaseURL: String, etTenant: String) -> Unit,
+    viewModel: SettingsViewModel = hiltViewModel(),
+    getSelectedLanguageIndex: () -> Int,
+    updateLanguage: (language: String) -> Unit,
+    getSelectedThemeIndex: () -> Int,
+    updateTheme: (selectedTheme: Int) -> Unit
+) {
+    val context = LocalContext.current
+
+    Scaffold(
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/settings/SettingsScreen.kt]: use MF SCaffold

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**